zebra/zapi_msg.c

```diff
@@ -2110,6 +2110,13 @@ static void zread_route_add(ZAPI_HANDLER_ARGS)
 	ret = rib_add_multipath_nhe(afi, api.safi, &api.prefix, src_p,
 				    re, &nhe);
 
+	/*
+	 * rib_add_multipath_nhe only fails in a couple spots
+	 * and in those spots we have not freed memory
+	 */
+	if (ret == -1)
```

Comment:
It looks like `-1` is treated as an `update` in the stats section a few lines further.
Just a notice in case you want to fix this also.

Reply:
I'm not sure what that code is trying to do as that it's just wrong and rib_add_multipath_nhe only ever returns -1 or 1 which makes no sense.  I'll fix in another commit

Reply:
Initially `rib_add_multipath` was returning `-1` for update, `1` for add and `0` for error. But then `-1` was mistakenly used as an error in https://github.com/FRRouting/frr/commit/98cda54a9543ea125e5e1eea6621c453f407edb2 and finally all other error returns were changed to `-1` in https://github.com/FRRouting/frr/commit/1d48702edeb418162940e51c99c942fc3c7bff77.
Actually the correct fix should be to change all those return values to be again `-1`, `0` and `1` for update, error and add.

Reply:
patience :) almost done with the fix.
walking the code to ensure it's right now